src/elements/transformer/Transform.ts

```diff
@@ -0,0 +1,36 @@
+import type { codeFields } from "../code/CodeElementSpec";
+import type { embedFields } from "../embed/EmbedSpec";
+import { transformElement as imageElementTransfrom } from "../image/ImageElementDataTransformer";
+import type { pullquoteFields } from "../pullquote/PullquoteSpec";
+import { transformElement as defaultElementTransform } from "./DefaultTransform";
+
+const transformMap = {
+  code: defaultElementTransform<typeof codeFields>(),
+  embed: defaultElementTransform<typeof embedFields>(),
+  image: imageElementTransfrom,
+  pullquote: defaultElementTransform<typeof pullquoteFields>(),
+} as const;
+
+type TransformMap = typeof transformMap;
+type TransformMapIn<Name extends keyof TransformMap> = TransformMap[Name][0];
+type TransformMapOut<Name extends keyof TransformMap> = TransformMap[Name][1];
+
+export const transfromElementIn = <Name extends keyof TransformMap>(
+  elementName: Name,
+  values: Parameters<TransformMapIn<Name>>[0]
+): ReturnType<TransformMapIn<Name>> => {
+  const transformer = transformMap[elementName][0];
```

Comment:
Can I recommend an object w/ `transformIn`, `transformOut` keys or similar rather than a tuple – just to make it clearer? Then we'd see e.g. `transformMap[elementName]["transformOut"]`, which I think is a little clearer.

Reply:
Good suggestion, I'll do that.